r2/r2/models/builder.py

```diff
@@ -109,6 +109,10 @@ def wrap_items(self, items):
                 if sr.can_ban(user):
                     can_ban_set.add(sr_id)
 
+        # hasattr check so no errors occur for non comments
+        links = Link._byID(set(l.link_id for l in items if hasattr(l, 'link_id')),
+                           data=True, return_dict=True, stale=self.stale)
```

Comment:
I used `stale=self.stale`, but in the current method via the `add_props`, `stale=True` no matter what. So you may want to double check this, just in case. Under casual observation I've had no issue.